src/mgxs.cpp

```diff
@@ -575,24 +573,20 @@ Mgxs::sample_fission_energy(int gin, int& dg, int& gout)
     // the neutron is delayed
 
     // get the delayed group
-    dg = 0;
-    while (xi_pd >= prob_prompt) {
-      dg++;
```

Comment:
Ooof. I'm guessing this was the bug? Probably easy to miss when looking for "- 1" to fix things after the C++/Fortran hybrid time period we had.

Reply:
Well, this is one of the bugs -- turns out this one leads to unphysical behavior (not sampling delayed neutrons correctly), but this is not actually what led to an out-of-bound index.